arcgis-ios-sdk-samples/Display information/Set up location-driven Geotriggers/SetUpLocationDrivenGeotriggersViewController.swift

```diff
@@ -0,0 +1,268 @@
+               let gardenSections = gardenSectionsLayer.featureTable as? AGSServiceFeatureTable,
+               let gardenPOIs = gardenPOIsLayer.featureTable as? AGSServiceFeatureTable {
+                // Create geotriggers for each of the service feature tables.
+                let geotriggerFeed = AGSLocationGeotriggerFeed(locationDataSource: locationDataSource)
+                self.createGeotriggerMonitor(feed: geotriggerFeed, featureTable: gardenSections, bufferDistance: 0.0, fenceGeotriggerName: self.sectionFenceGeotriggerName)
+                self.createGeotriggerMonitor(feed: geotriggerFeed, featureTable: gardenPOIs, bufferDistance: 10.0, fenceGeotriggerName: self.poiFenceGeotriggerName)
+            }
+        }
+        mapView.map = map
+    }
+    
+    func setupLocationDisplay(locationDataSource: AGSLocationDataSource) {
+        mapView.locationDisplay.dataSource = locationDataSource
+        mapView.locationDisplay.autoPanMode = .recenter
+        mapView.locationDisplay.initialZoomScale = 1000
+        mapView.locationDisplay.start()
+    }
+    
+    func updateStatusLabels(featureName: String, notificationType: AGSFenceNotificationType) {
+        let typeString = notificationType == .entered ? "Entered" : "Exited"
+        let fenceGeotriggerText = String(format: "%@ the geofence of %@", typeString, featureName)
+        let nearbyFeaturesText: String
+        if nearbyFeatures.keys.isEmpty {
+            nearbyFeaturesText = "No nearby features."
+        } else {
+            nearbyFeaturesText = String(format: "Nearby: %@", ListFormatter.localizedString(byJoining: Array(nearbyFeatures.keys)))
+        }
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            self.fenceGeotriggerLabel.text = fenceGeotriggerText
+            self.nearbyFeaturesLabel.text = nearbyFeaturesText
+        }
+    }
+    
+    func showPopups(for features: [AGSArcGISFeature]) {
+        let popups = features.map { AGSPopup(geoElement: $0) }
+        let popupsViewController = AGSPopupsViewController(popups: popups)
+        popupsViewController.delegate = self
+        // Display the view controller as a formsheet - specified for iPads.
+        popupsViewController.modalPresentationStyle = .formSheet
+        popupsViewController.isModalInPresentation = true
+        // Present the popups view controller.
+        present(popupsViewController, animated: true) {
+            // Stop location updates when it is presented.
+            self.mapView.locationDisplay.stop()
+        }
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func currentSectionButtonTapped(_ sender: UIBarButtonItem) {
+        if let section = currentSectionName,
+           let sectionFeature = nearbyFeatures[section] {
+            showPopups(for: [sectionFeature])
+        } else {
+            presentAlert(title: "Not in a Garden Section", message: "Currently we are not in a garden section.")
```

Comment:
Since the buttons' enabled state are determined by the emptiness of the dict collections, this should never happen.

However I'm afraid of thread timing bug when UI updates are dispatched onto main queue from the notification's queue, so I added these alerts here. 

Should I just force unwrap the variables above?